pages/resources/quizzes/FirstLang/index.tsx

```diff
@@ -0,0 +1,107 @@
+import { useState } from "react";
+import LangResults from "../../../../components/resources/quizzes/langQuiz/LangResults";
+import BluePrint from "../../../../components/resources/quizzes/shared/BluePrint";
+import SkillSelections from "../../../../components/resources/quizzes/shared/SkillSelections";
+import StartQuiz from "../../../../components/resources/quizzes/shared/StartQuiz";
+import {
+  quizQuestions,
+  selectionsPerPage,
+} from "../../../api/studentPortal/quizzes/FirstLang";
+
+export enum Stage {
+  START,
+  LEARNCODING,
+  WORKFIELDS,
+  BUILDTARGET,
+  BLUEPRINT,
+  RESULTS,
+}
+
+const FirstProgrammingLanguageQuiz = () => {
+  const [selectedSkill, setSelectedSkill] = useState([]);
+
+  const [stage, setStage] = useState(Stage.START);
+
+  const handleNextClick = () => {
+    setStage((prevStage) => prevStage + 1);
+  };
+  const handleBackClick = () => {
+    setStage((prevStage) => prevStage - 1);
+  };
+
+  const renderStage = () => {
+    switch (stage) {
+      case Stage.START:
+        return (
+          <StartQuiz
+            onNextClick={handleNextClick}
+            title={quizQuestions[stage][stage].title}
+            body={quizQuestions[stage][stage].body}
+          />
+        );
+      case Stage.LEARNCODING:
+        return (
+          <SkillSelections
+            onNextClick={handleNextClick}
+            onBackClick={handleBackClick}
+            title={quizQuestions[stage][stage - 1].title}
```

Comment:
this is not great either, you are assuming that the title you are looking for will always be at the top of the array, which isn't guaranteed. how will you prevent a new team member from accidentally adding a question to the top of the array?